packages/wrangler/src/publish.ts

```diff
@@ -239,135 +261,142 @@ export default async function publish(props: Props): Promise<void> {
       usage_model: config.usage_model,
     };
 
-    const start = Date.now();
-    const notProd = !props.legacyEnv && props.env;
-    const workerName = notProd ? `${scriptName} (${envName})` : scriptName;
-    const workerUrl = notProd
-      ? `/accounts/${accountId}/workers/services/${scriptName}/environments/${envName}`
-      : `/accounts/${accountId}/workers/scripts/${scriptName}`;
-
-    // Upload the script so it has time to propagate.
-    const { available_on_subdomain } = await fetchResult(
-      workerUrl,
-      {
-        method: "PUT",
-        body: createWorkerUploadForm(worker),
-      },
-      new URLSearchParams({ available_on_subdomain: "true" })
-    );
-
-    const uploadMs = Date.now() - start;
-    const deployments: Promise<string[]>[] = [];
-
-    if (deployToWorkersDev) {
-      // Deploy to a subdomain of `workers.dev`
-      const userSubdomain = await getSubdomain(accountId);
-      const scriptURL =
-        props.legacyEnv || !props.env
-          ? `${scriptName}.${userSubdomain}.workers.dev`
-          : `${envName}.${scriptName}.${userSubdomain}.workers.dev`;
-      if (!available_on_subdomain) {
-        // Enable the `workers.dev` subdomain.
-        deployments.push(
-          fetchResult(`${workerUrl}/subdomain`, {
-            method: "POST",
-            body: JSON.stringify({ enabled: true }),
-            headers: {
-              "Content-Type": "application/json",
-            },
-          })
-            .then(() => [scriptURL])
-            // Add a delay when the subdomain is first created.
-            // This is to prevent an issue where a negative cache-hit
-            // causes the subdomain to be unavailable for 30 seconds.
-            // This is a temporary measure until we fix this on the edge.
-            .then(async (url) => {
-              await sleep(3000);
-              return url;
-            })
-        );
-      } else {
-        deployments.push(Promise.resolve([scriptURL]));
-      }
-    } else {
-      // Disable the workers.dev deployment
-      await fetchResult(`${workerUrl}/subdomain`, {
-        method: "POST",
-        body: JSON.stringify({ enabled: false }),
-        headers: {
-          "Content-Type": "application/json",
-        },
-      });
+    if (!props.dryRun) {
+      // Upload the script so it has time to propagate.
+      // We can also now tell whether available_on_subdomain is set
+      available_on_subdomain = (
+        await fetchResult<{ available_on_subdomain: boolean }>(
+          workerUrl,
+          {
+            method: "PUT",
+            body: createWorkerUploadForm(worker),
+          },
+          new URLSearchParams({ available_on_subdomain: "true" })
+        )
+      ).available_on_subdomain;
     }
+  } finally {
+    if (typeof destination !== "string") {
+      // this means we're using a temp dir,
+      // so let's clean up before we proceed
+      await destination.cleanup();
+    }
+  }
```

Comment:
I don't think you needed to move this finally block inside, right?
Leaving it where it was would still get called, I think, and reduce the amount of changes near here. But you might have already tried that and I might be missing something.

Reply:
I decided to move this up so that it would do a strict enforce of not depending on the built assets after this step. It doesn't look like a lot of changes if you remove whitespace.

Reply:
Not sure, I follow, but it is not a blocker for me.